### PR TITLE
Add/search block tests

### PR DESCRIPTION
**Pull request by sowmya2709**
#### Changes proposed in this Pull Request

*Added new tests to validate functionality of Gutenberg Search block.  - test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js

To support the above test, added Search related 'case' in this library file -test/e2e/lib/gutenberg/gutenberg-editor-component.js

@bsessions85 @worldomonation @kwight  - Some of the other changes to config files were made when I was trying to decrypt and run the test. Please let me know if they are need to be reverted, I will do the same and create another PR.

Regards,
Sowmya

#### Testing instructions

*

Related to #

### CODE REVIEW

**Comment by bsessions85** [comment]
@sowmya2709 Yes, if you could clean up all the config and other changes that don't belong in here that would be great. It doesn't have to be a new PR, just change them back.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+			return driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button' )
+			);
+		} );
+
+		it( 'Can publish and view content', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			// We need to save the post to get a stable post slug for the block's `url` attribute.
+			// See https://github.com/godaddy-wordpress/coblocks/issues/1663.
```
Comment (bsessions85) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: This seems to be only for a specific block.  I don't think this comment applies here.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.addBlock( 'Search' );
+			return await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search' )
+			);
+		} );
+
+		it( 'Can Toggle Search Block', async function () {
+			const textLocator = By.css( '.wp-block-search__input' );
```
Comment (bsessions85) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: A lot of the settings in this section might work better as a "Page Object" that contains the settings for the search block. It would really clean this up and let you re-use things.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			await driver.findElement( labelToggleButton ).click();
+
+			return driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button.has-icon' )
+			);
+		} );
+
+		it( 'Can remove search button', async function () {
+			//html/body/div[1]/div[2]/div[1]/div[1]/div[3]/div[1]/div[2]/div/div/div/div/div/div/button[3]
```
Comment (bsessions85) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Can this be removed?

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+		it( 'Can see the 25% Search block in published post without label and button', async function () {
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__label' ) );
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__button' ) );
+
+			//Unable to get width of the search block here and neither in edit mode. In edit mode it gets the  min-width:290px
+			//Here in publish mode it get 173px though that style value is not there when I inspect the object
+			const width = await driver
+				.findElement( By.css( '.wp-block-search__inside-wrapper' ) )
+				.getAttribute( 'style' );
+			console.log( 'Style Value' + width );
```
Comment (bsessions85) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: This should be removed

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+			await gEditorComponent.ensureSaved();
+			return await gEditorComponent.publish( { visit: true } );
+		} );
+
+		it( 'Can see the 25% Search block in published post without label and button', async function () {
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__label' ) );
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__button' ) );
+
+			//Unable to get width of the search block here and neither in edit mode. In edit mode it gets the  min-width:290px
+			//Here in publish mode it get 173px though that style value is not there when I inspect the object
```
Comment (bsessions85) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Just curious what `getRect()`got you since it is part of the Selenium 4 spec. I know these size things can vary greatly so it isn't surprising that it wasn't giving you what you expected.

**Review by bsessions85 — COMMENTED** [review]
Beyond just cleaning up the things that shouldn't have changed, I left some comments.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 4 above)
Comment (sowmya2709) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Looks like I committed without saving my last sublime edit :)  Re- Commiting

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 5 above)
Comment (sowmya2709) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: getRect() - This gives height and width  in pixels along with coordinates. It requires mapping 25% to 173, which I felt is not directly understandable. So wanted to use the "width: 25%;" style property directly. So used getAttribute instead as it's self-explanatory.  Here is what I get 

 + {
+   height: 57,
+   width: 173,
+   x: 202.5,
+   y: 347.5
+ }

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 2 above)
Comment (sowmya2709) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: @bsessions85 - I will add these to Page objects. Just confirming, this is where all the page objects are right? /wp-calypso/test/e2e/lib/pages

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 5 above)
Comment (bsessions85) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Ok, that makes sense. I was just curious. 😃

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 2 above)
Comment (bsessions85) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Yeah, they are all in lib somewhere. Blocks might be better in gutenberg/blocks

**Comment by worldomonation** [comment]
I've triggered tests on TeamCity for this branch. Will review the PR in depth some more later.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 2 above)
Comment (sowmya2709) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: @bsessions85  - Is there any README page that gives more information about how the Page Objects set up in calypso . I was trying to read this https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/docs/style-guide.md#page-objects and follow up but what's mentioned in "Don't Do" , some flows/components do that.

Comment (bsessions85) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: @sowmya2709 That is really all the documentation that we have. The key in that readme section is to make sure you're using the 'expect' or 'visit' method from the test and not just creating a new instance of the object. I recommend finding one in there and using it as a template.

File: client/lib/plans/features-list.js [hunk 6]
```diff
@@ -42,7 +42,7 @@ import {
 	FEATURE_BLANK,
 	FEATURE_BLOG_DOMAIN,
 	FEATURE_BUSINESS_ONBOARDING,
-  FEATURE_CLOUDFLARE_ANALYTICS,
+	FEATURE_CLOUDFLARE_ANALYTICS,
```
Comment (worldomonation) [inline client/lib/plans/features-list.js:45]: nit: please ensure these kinds of unrelated changes do not make it into the PR.

File: package.json [hunk 7]
```diff
@@ -332,4 +332,4 @@
 		"chokidar": "^3.5.1",
 		"eslint-nibble": "^6.1.0"
 	}
-}
\ No newline at end of file
+}
```
Comment (worldomonation) [inline package.json]: nit: same here as the `client/lib/plans/features-list.js` file.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 2 above)
Comment (worldomonation) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: The documents are present but there is no guarantee that they are up to date. The code itself, and how it's structured should be the primary form of documentation.

Take a look at some of the page objects that exist in the repo for inspiration. A good one is [posts-page](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/pages/posts-page.js) and another is [login-page](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/pages/login-page.js).

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+		it( 'Can remove search button', async function () {
+			const textLocator = By.css( 'button[aria-label="Change button position"]' );
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, textLocator );
+			await driver.findElement( textLocator ).click();
+			await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button-position-menu' )
+			);
+			const buttonEl = await driver.findElement( By.xpath( `//span[text()='No Button']` ) );
```
Comment (worldomonation) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: nit: It would be preferable to use names like`button` or `buttonElement`.

**Comment by worldomonation** [comment]
I pulled the branch locally and was able to verify that it runs as expected. The tests are also laid out 

I did have some thoughts relating to this specific PR:

1. the test checks if the block is present, but the acceptance test criteria also mentions elements search block and its elements. Things like the placeholder text may be a valuable element to check for.
2. have you tried entering text into the search block? If so, was it successful or was something else a blocker?

Other than that, interested in seeing the other PR for Pricing Table menu block.

**Comment by sowmya2709** [comment]
Hi Edwin,

Thanks for the feedback.
1) Regarding this point, I agree checking for placeholder text will be a
valuable element to add and also check that on the published page.  I did
add in the manual test and somehow missed that one.
2) When I tested manually I created a post and searched it on the website
to make sure it works. I think I could have at least included the "No Post
Found" test in the Search block without adding other blocks etc.
 However, while automating and running the tests for a new account and
checking for just the search block, I had doubts about adding a Post block
in the search spec and then searching that.
So did not include that in one of the 5 tests to automate. Having said
that, it is one of the important functional validation of the Search block
and test that searches for something that exists and does not exist should
both be part of Search block automation tests.

I was planning to write up a P2 this.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js (hunk 2 above)
Comment (sowmya2709) [reply test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js]: Hi @worldomonation, Thanks for these references

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
+/**
+ * External dependencies
```
Comment (sowmya2709) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js:2]: @worldomonation @bsessions85   -This is the WIP test case for pricing. Please ignore this one for now.

File: test/e2e/config/development.json [hunk 10]
```diff
@@ -1,4 +1,4 @@
 {
-	"calypsoBaseURL": "http://calypso.localhost:3000",
+	"calypsoBaseURL": "https://wordpress.com",
```
Comment (kwight) [inline test/e2e/config/development.json]: Is this change necessary @sowmya2709 ?

Comment (sowmya2709) [reply test/e2e/config/development.json]: @kwight  -For some reason, I am not able to run calypso from my machine. So changed this, I will remove it in my final commit

File: client/lib/plans/features-list.js (hunk 6 above)
Comment (sowmya2709) [reply client/lib/plans/features-list.js:45]: @worldomonation - I did not make changes to these files deliberately, it could have happened when I was trying to set up test env. I was wondering if it would be ok to delete this file from the fork?

File: package.json (hunk 7 above)
Comment (sowmya2709) [reply package.json]: @worldomonation  - I tried checking out the latest from package.json from trunk to overwrite my changes but it looks like when I did a "git pull origin trunk" these changes got into my fork. I am not able to  revert these file back to the state they were when i forked the branch
